nix/haskell-project.nix

```
@@ -55,14 +58,38 @@ in
                   hpack
                 ''
             else root;
+          makeSrcAutonomous = name: root: pkg: hlib.overrideSrc
+            {
+              src =
+                # Since 'root' maybe a subdirectory of another source
+                # derivation, the purpose of cleanSourceWith here is to create a
+                # new (smaller) source derivation that is a copy of 'root' but
+                # ignores the parent source contents.
```

```suggestion
                # Since 'root' may be a subdirectory of a store path
                # (in string form, which means that it isn't automatically
                # copied), the purpose of cleanSourceWith here is to create a
                # new (smaller) store path that is a copy of 'root' but
                # does not contain the unrelated parent source contents.
```

It wasn't a derivation output at this point yet.

---

Thanks; done.